doc/sphinx/README.rst

```diff
@@ -0,0 +1,298 @@
+          ``assert``.
+
+          .. exn:: Proof is not complete
+
+             Raised is :n:`@tactic` does not fully solve the goal.
+
+``.. opt::`` :black_nib: A Coq option.
+    Example::
+
+       .. opt:: Nonrecursive Elimination Schemes
+
+          This option controls whether types declared with the keywords
+          :cmd:`Variant` and :cmd:`Record` get an automatic declaration of the
+          induction principles.
+
+``.. prodn::`` :black_nib: Grammar productions.
+    This is useful if you intend to document individual grammar productions.
+    Otherwise, use Sphinx's `production lists
+    <http://www.sphinx-doc.org/en/stable/markup/para.html#directive-productionlist>`_.
+
+``.. tacn::`` :black_nib: A tactic, or a tactic notation.
+    Example::
+
+       .. tacn:: do @num @expr
+
+          :token:`expr` is evaluated to ``v`` which must be a tactic value. …
+
+``.. tacv::`` :black_nib: A variant of a tactic.
+    Example::
+
+       .. tacn:: fail
+
+          This is the always-failing tactic: it does not solve any goal. It is
+          useful for defining other tacticals since it can be caught by
+          :tacn:`try`, :tacn:`repeat`, :tacn:`match goal`, or the branching
+          tacticals. …
+
+          .. tacv:: fail @natural
+
+             The number is the failure level. If no level is specified, it
+             defaults to 0. …
+
+``.. thm::`` A theorem.
+    Example::
+
+       .. thm:: Bound on the ceiling function
+
+          Let :math:`p` be an integer and :math:`c` a rational constant. Then
+          :math:`p \ge c \rightarrow p \ge \lceil{c}\rceil`.
+
+``.. warn::`` :black_nib: An warning raised by a Coq command or tactic..
+    Do not mistake this for ``.. warning::``; this directive is for warning
+    messages produced by Coq.
+
+
+    Example::
+
+       .. warn:: Ambiguous path
+
+         When the coercion `qualid` is added to the inheritance graph, non
+         valid coercion paths are ignored.
+
+Coq directives
+==============
+
+In addition to the objects above, the ``coqrst`` Sphinx plugin defines the following directives:
+
+``.. coqtop::`` A reST directive to describe interactions with Coqtop.
+    Usage::
+
+       .. coqtop:: options…
+
+          Coq code to send to coqtop
+
+    Example::
+
+       .. coqtop:: in reset undo
+
+          Print nat.
+          Definition a := 1.
+
+    Here is a list of permissible options:
+
+    - Display options
+
+      - ``all``: Display input and output
+      - ``in``: Display only input
+      - ``out``: Display only output
+      - ``none``: Display neither (useful for setup commands)
+
+    - Behavior options
+
+      - ``reset``: Send a ``Reset Initial`` command before running this block
+      - ``undo``: Send an ``Undo n`` (``n`` = number of sentences) command after
+        running all the commands in this block
+
+``.. coqdoc::`` A reST directive to display Coqtop-formatted source code.
```

Comment:
Didn't know about this one. It seems that lots of places in the current manual are using `.. coqtop:: in` instead.

Reply:
That should look the same, though it's likely a bit slower.